src/components/Tutorial/NavigationBar.vue

```diff
@@ -117,11 +126,26 @@ export default {
   },
   computed: {
     pageSections() {
-      const tutorialPath = this.$route.path.replace(/#$/, '');
-      return this.tutorialState.linkableSections.map(section => ({
-        ...section,
-        path: `${tutorialPath}#${section.anchor}`,
-      }));
+      const projectReferences = this.chapters
+        .reduce((acc, { projects }) => acc.concat(projects), [])
+        .find(project => project.reference === this.identifierUrl);
+      if (!projectReferences) return [];
+
+      const allSections = [FirstSectionItem, ...projectReferences.sections];
+      const { linkableSections } = this.tutorialState;
+
+      return linkableSections.map((linkableSection, index) => {
+        const singleSection = allSections[index];
+        const referenceObject = this.references[singleSection.reference];
+        const { url, title } = referenceObject || singleSection;
+        return {
+          depth: linkableSection.depth,
+          visibility: linkableSection.visibility,
+          sectionNumber: linkableSection.sectionNumber,
+          title,
+          path: url,
```

Comment:
Is url a required "key" for the renderjson? If not, should we keep the old logic to make sure if no url is provided, we fall back to use the anchor?